addons/todo/models/project_task.py

```diff
@@ -21,6 +21,34 @@ def create(self, vals_list):
                 vals['name'] = _('Untitled to-do')
         return super().create(vals_list)
 
+    def _ensure_onboarding_todo(self):
+        if not self.env.user.has_group('todo.group_onboarding_todo'):
+            self._generate_onboarding_todo(self.env.user)
+            onboarding_group = self.env.ref('todo.group_onboarding_todo').sudo()
+            onboarding_group.users += self.env.user
+
+    def _generate_onboarding_todo(self, users):
+        todos_to_create = []
+        for user in users:
+            body = self.with_context(lang=user.lang or self.env.user.lang).env['ir.qweb']._render(
+                'todo.todo_user_onboarding',
+                {'object': user},
+                minimal_qcontext=True,
+                raise_if_not_found=False
+            )
+            if not body:
+                break
+
+            title = _lt('Welcome %s!', user.name)
+            todos_to_create.append({
+                'user_ids': user.ids,
+                'description': body,
+                'name': title,
+            })
+
+        if todos_to_create:
+            self.env['project.task'].sudo().with_context(onboarding_todo_creation=True).create(todos_to_create)
```

Comment:
The context key is not longer needed since now the method will only be used when the user will go for the first time in the todo app, right? 🤔

If the context key is really needed then we could rename it to have something more generic, like `self_assign`? 🤔

Reply:
True, as it is now created per user basis, we don't need the context key anymore and keeping the possibility to generated onboarding todos in batch is not really needed. I've changed that part.